movr.py

```diff
@@ -15,44 +17,46 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_value, traceback):
         self.session.close()
 
-    def __init__(self, conn_string, init_tables = False, multi_region = False, echo = False):
+    def __init__(self, conn_string, reset_tables=False, multi_region=False, primary_region=None, echo=False):
 
+        self.engine = create_engine(
+            conn_string, convert_unicode=True, echo=echo)
+        self.session = sessionmaker(bind=self.engine)()
+        self.primary_region = primary_region
         self.multi_region = multi_region
-        self.engine = create_engine(conn_string, convert_unicode=True, echo=echo)
 
-
-        if init_tables:
+        if reset_tables:
+            logging.info("reseting movr schema")
             logging.info("initializing tables")
             Base.metadata.drop_all(bind=self.engine)
             Base.metadata.create_all(bind=self.engine)
+            logging.debug("tables dropped and created")
             if multi_region:
                 self.run_multi_region_transformations()
 
-
-            logging.debug("tables dropped and created")
-
-        self.session = sessionmaker(bind=self.engine)()
-
     ##################
     # MAIN MOVR API
     #################
 
     def start_ride(self, city, rider_id, vehicle_id):
 
         def start_ride_helper(session, city, rider_id, vehicle_id):
-            vehicle = session.query(Vehicle).filter_by(city=city, id=vehicle_id) if self.multi_region else session.query(Vehicle).filter_by(id=vehicle_id)
-            vehicle.update({'status': 'in_use' })
+            vehicle = session.query(Vehicle).filter_by(
+                city=city, id=vehicle_id) if self.multi_region else session.query(Vehicle).filter_by(id=vehicle_id)
```

Comment:
Do we still need to filter by city now that the id is the primary key? this just looks like a point lookup now for single and multi-region cases. I think city is just for the initial cache warming

Reply:
this pattern shows up below as well

Reply:
interesting - is that union happening in parallel though? if so that's OK IMO.

@awoods187 what is your take here? I thought we could still do point lookups by PK and we checked all partitions in parallel? I'm surprised we have to include "city" here. If we do, then I think we need to preserve the multi-region option for queries so we don't have to include it in the single region case.

Reply:
This is the correct behavior. When you don't specify the region with the where predicate clause, first we check the local gateway region (we assume the app connecting to the gateway is from the same region), then if that's not found, we check the other regions in parallel. This is to not require the app to pass the region information or city information back to the db. Of course if you know the region there is no harm in it but this shouldn't be a problem. If the data is local, you get a ~2ms read, if its not local, the extra ~2ms to check local is trivial when compared to the latency of traveling to other regions. The beauty of all of this is that you don't need to change the queries when going from the single region to the multi region version.

Reply:
Thanks for that summary @awoods187 !

I've updated the DML queries to not filter on `city`.